02-peer-protocol.md

```diff
@@ -445,6 +446,203 @@ to broadcast the commitment transaction to get his funds back and open a new
+- MUST pay for the common fields.
+
+Upon receipt of consecutive `tx_complete`s, each node:
+- MUST fail negotiation if there is not exactly one input spending the current funding transaction.
+- MUST fail negotiation if there is not exactly one output with zero value paying to the two funding keys (a.k.a. the new channel funding output)
+- MUST calculate the channel capacity for each side:
+  - Start with the previous balance
+  - Add that side's new inputs (excluding the one spending the current funding transaction)
+  - Subtracting each sides new outputs (except the zero-value one paying to the funding keys)
+  - Subtract the total fee that side is paying for the splice transaction.
+- MUST replace the zero-value funding output amount with the total channel capacity.
+- MUST calculate the channel balance for each side:
+  - Subtract any outstanding HTLCs offered by that side.
+- If either side has added an output other than the new channel funding output:
+  - MUST fail the negotiation if the balance for that side is less than 1% of the total channel capacity.
+- SHOULD NOT fail if the splice transaction is nonstandard.
+- MUST send a single `commitment_signed` for the splice transaction.
+  - Peer MUST NOT reply with `revoke_and_ack`
+
+- If recipient's sum(tx_add_input.amount) < peer's sum(tx_add_input.amount); or
+if recipient's sum(tx_add_input.amount) == peer's sum(tx_add_input.amount) and
+recipient is the `initiator` of the splice:
+  - SHOULD send `tx_signatures` first for the splice transaction.
+- else
+  - MAY send `tx_signatures` first.
+
+- Upon receipt of `tx_signatures` for the splice transaction:
+  - MUST consider splice negotiation complete.
+  - MUST consider the connection no longer quiescent.
+
+On reconnection:
+- MUST retransmit the last splice `tx_signatures` (if any).
+- MUST ignore any redundant `tx_signatures` it receives.
+
+### Rationale
+
+If a side does not meet the reserve requirements, that's OK: but if
+they take funds out of the channel, they must ensure that they do meet
+them.  If your peer adds a massive amount to the channel, then you
+only have to add more reserve if you want to contribute to the splice
+(and you can use `tx_remove_output` and/or `tx_remove_input` part-way
+through if this happens).
+
+## Splice Completion
+
+### The `splice_locked` Message
+
+1. type: 77 (`splice_locked`)
+2. data:
+   * [`channel_id`:`channel_id`]
+
+`channel_id` is the pre-splice channel's `channel_id`.
+
+Each node:
+- if any splice transaction reaches depth 6:
+  - MUST send `splice_locked`.
+
+Once a node has received and sent `splice_locked`:
+  - Until sending OR receiving of `revoke_and_ack`
+    - MUST ignore `announcement_signatures` messages where `short_channel_id`
+      matches the pre-splice short channel id.
+    - MUST ignore `commitment_signed` messages where `splice_channel_id`
+      does not match the `channel_id` of the confirmed splice.
```

Comment:
Can we more explicitly define this `channel_id`? Is it just TXID based as usual? So even if we started with a dual-funded channel, where `channel_id` is XORed revocation basepoints, then after successful splice we now have a TXID-based one?